src/client/auth.rs

```diff
@@ -470,15 +471,22 @@ pub struct Credential {
     pub mechanism_properties: Option<Document>,
 
     /// The token callback for OIDC authentication.
-    // TODO RUST-1497: make this `pub`
-    // Credential::builder().oidc_callback(oidc::Callback::human(...)).build()
-    // the name of the field here does not well encompass what this field actually is since
-    // it contains all the OIDC state information, not just the callback, but it conforms
-    // to how a user would interact with it.
+    /// ```
+    /// let mut opts =
+    /// ClientOptions::parse("mongodb://localhost:27017,localhost:27018/admin?authSource=admin&authMechanism=MONGODB-OIDC").await?;
+    /// opts.credential.as_mut().unwrap().oidc_callback =
+    ///     Callback::human(move |c: CallbackContext| {
+    ///     async move {
+    ///         let (access_token, expires, refresh_token) = do_human_flow(c).await?;
+    ///         Ok(IdpServerResponse::builder().access_token(access_token).expires(expires).refresh_token(refresh_token).build())
+    ///     }.boxed()
+    /// });
+    /// ```
```

Comment:
using private interfaces because we do not want users creating `State` other than with the `human` or `machine` function.